paddle/contrib/inference/paddle_inference_api_anakin_engine.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda.h>
+
+#include "paddle/contrib/inference/paddle_inference_api_anakin_engine.h"
+
+namespace paddle {
+
+PaddleInferenceAnakinPredictor::PaddleInferenceAnakinPredictor(
+    const AnakinConfig &config) {
+  CHECK(Init(config));
+}
+
+bool PaddleInferenceAnakinPredictor::Init(const AnakinConfig &config) {
+  // TODO(Superjomn) Tell anakin to support return code.
+  engine_.Build(config.model_file, config.max_batch_size);
+  return true;
+}
+
+bool PaddleInferenceAnakinPredictor::Run(
+    const std::vector<PaddleTensor> &inputs,
+    std::vector<PaddleTensor> *output_data) {
+  for (const auto &input : inputs) {
+    CHECK(input.dtype == PaddleDType::FLOAT32);
+    engine_.SetInputFromCPU(
+        input.name, static_cast<float *>(input.data.data), input.data.length);
+  }
+
+  // TODO(Superjomn) Tell anakin to support return code.
+  engine_.Execute();
+
+  CHECK(!output_data->empty())
```

Comment:
maybe not need to worry about it?

Reply:
I found some issue about the usage of the `output_data`.  The current implementation for the native engine is that the `output_data`'s memory is maintained by the engine, but the original intention is to force the users to maintain the memory outside and the engine just copies the data.

So that, it can support

- fetch any target, just set the PaddleTensor's names of `output_data` -- given a big model, fetch any variable
- engines like Anakin's output variables are located at GPU, need a memory to copied, and users can do some memory optimization such as pre-allocate a big buffer.